### ISSUE TITLE
bug: Port could not be cast to integer value as {port!r}

### ISSUE THREAD

**Issue by Isanderthul** [localstack/localstack#11900]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

when running localstack with scripts from my company, I get this error
```
 File "/usr/local/lib/python3.11/urllib/parse.py", line 182, in port
    raise ValueError(f"Port could not be cast to integer value as {port!r}")
ValueError: Port could not be cast to integer value as ' 4566'
```

### Expected Behavior

It should support the port number for the sqs url

### How are you starting LocalStack?

Custom (please describe below)

### Steps To Reproduce

custom shell script

### Environment

```markdown
- OS: mac
- LocalStack:latest 
  LocalStack version:
  LocalStack Docker image sha256:647b912d48c6fdcba0e035cc0bc1002d5c173df179a34afa85deaf0ef680ba66
  LocalStack build date:
  LocalStack build git hash:
44	ENV LOCALSTACK_BUILD_DATE=2024-11-20	0 B
	45	ENV LOCALSTACK_BUILD_GIT_HASH=366078400	0 B
	46	ENV LOCALSTACK_BUILD_VERSION=3.8.2.dev144	0 B
```

### Anything else?

```
➜  fleet-compose git:(master) AWS_PROFILE=us1-admin fleet start
Starting ALL containers...
Waiting for LocalStack
localstack is ready

An error occurred (InternalError) when calling the CreateStream operation (reached max retries: 2): exception while calling kinesis.CreateStream: Traceback (most recent call last):
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/rolo/gateway/chain.py", line 166, in handle
    handler(self, self.context, response)
  File "/opt/code/localstack/localstack-core/localstack/aws/handlers/service.py", line 113, in __call__
    handler(chain, context, response)
  File "/opt/code/localstack/localstack-core/localstack/aws/handlers/service.py", line 83, in __call__
    skeleton_response = self.skeleton.invoke(context)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/localstack-core/localstack/aws/skeleton.py", line 154, in invoke
    return self.dispatch_request(serializer, context, instance)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/localstack-core/localstack/aws/skeleton.py", line 168, in dispatch_request
    result = handler(context, instance) or {}
             ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/localstack-core/localstack/aws/forwarder.py", line 140, in _call
    return fallthrough_handler(context, req)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/localstack-core/localstack/aws/forwarder.py", line 164, in _forward_request
    return AwsRequestProxy(forward_url_getter(context.account_id, context.region)).forward(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/localstack-core/localstack/aws/forwarder.py", line 77, in forward
    http_response = self.proxy.forward(context.request, forward_path=context.request.path)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/rolo/proxy.py", line 83, in forward
    proxy_request = _copy_request(request, self.forward_base_url, forward_path, headers)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/rolo/proxy.py", line 149, in _copy_request
    builder = EnvironBuilder.from_environ(request.environ)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/werkzeug/test.py", line 427, in from_environ
    return cls(**out)
           ^^^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/werkzeug/test.py", line 328, in __init__
    base_url = iri_to_uri(base_url)
               ^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/werkzeug/urls.py", line 153, in iri_to_uri
    if parts.port:
       ^^^^^^^^^^
  File "/usr/local/lib/python3.11/urllib/parse.py", line 182, in port
    raise ValueError(f"Port could not be cast to integer value as {port!r}")
ValueError: Port could not be cast to integer value as ' 4566'


An error occurred (InternalError) when calling the CreateStream operation (reached max retries: 2): exception while calling kinesis.CreateStream: Traceback (most recent call last):
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/rolo/gateway/chain.py", line 166, in handle
    handler(self, self.context, response)
  File "/opt/code/localstack/localstack-core/localstack/aws/handlers/service.py", line 113, in __call__
    handler(chain, context, response)
  File "/opt/code/localstack/localstack-core/localstack/aws/handlers/service.py", line 83, in __call__
    skeleton_response = self.skeleton.invoke(context)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/localstack-core/localstack/aws/skeleton.py", line 154, in invoke
    return self.dispatch_request(serializer, context, instance)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/localstack-core/localstack/aws/skeleton.py", line 168, in dispatch_request
    result = handler(context, instance) or {}
             ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/localstack-core/localstack/aws/forwarder.py", line 140, in _call
    return fallthrough_handler(context, req)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/localstack-core/localstack/aws/forwarder.py", line 164, in _forward_request
    return AwsRequestProxy(forward_url_getter(context.account_id, context.region)).forward(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/localstack-core/localstack/aws/forwarder.py", line 77, in forward
    http_response = self.proxy.forward(context.request, forward_path=context.request.path)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/rolo/proxy.py", line 83, in forward
    proxy_request = _copy_request(request, self.forward_base_url, forward_path, headers)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/rolo/proxy.py", line 149, in _copy_request
    builder = EnvironBuilder.from_environ(request.environ)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/werkzeug/test.py", line 427, in from_environ
    return cls(**out)
           ^^^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/werkzeug/test.py", line 328, in __init__
    base_url = iri_to_uri(base_url)
               ^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/werkzeug/urls.py", line 153, in iri_to_uri
    if parts.port:
       ^^^^^^^^^^
  File "/usr/local/lib/python3.11/urllib/parse.py", line 182, in port
    raise ValueError(f"Port could not be cast to integer value as {port!r}")
ValueError: Port could not be cast to integer value as ' 4566'


An error occurred (InternalError) when calling the ListStreams operation (reached max retries: 2): exception while calling kinesis.ListStreams: Traceback (most recent call last):
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/rolo/gateway/chain.py", line 166, in handle
    handler(self, self.context, response)
  File "/opt/code/localstack/localstack-core/localstack/aws/handlers/service.py", line 113, in __call__
    handler(chain, context, response)
  File "/opt/code/localstack/localstack-core/localstack/aws/handlers/service.py", line 83, in __call__
    skeleton_response = self.skeleton.invoke(context)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/localstack-core/localstack/aws/skeleton.py", line 154, in invoke
    return self.dispatch_request(serializer, context, instance)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/localstack-core/localstack/aws/skeleton.py", line 168, in dispatch_request
    result = handler(context, instance) or {}
             ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/localstack-core/localstack/aws/forwarder.py", line 140, in _call
    return fallthrough_handler(context, req)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/localstack-core/localstack/aws/forwarder.py", line 164, in _forward_request
    return AwsRequestProxy(forward_url_getter(context.account_id, context.region)).forward(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/localstack-core/localstack/aws/forwarder.py", line 77, in forward
    http_response = self.proxy.forward(context.request, forward_path=context.request.path)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/rolo/proxy.py", line 83, in forward
    proxy_request = _copy_request(request, self.forward_base_url, forward_path, headers)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/rolo/proxy.py", line 149, in _copy_request
    builder = EnvironBuilder.from_environ(request.environ)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/werkzeug/test.py", line 427, in from_environ
    return cls(**out)
           ^^^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/werkzeug/test.py", line 328, in __init__
    base_url = iri_to_uri(base_url)
               ^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/werkzeug/urls.py", line 153, in iri_to_uri
    if parts.port:
       ^^^^^^^^^^
  File "/usr/local/lib/python3.11/urllib/parse.py", line 182, in port
    raise ValueError(f"Port could not be cast to integer value as {port!r}")
ValueError: Port could not be cast to integer value as ' 4566'

make_bucket: local-media
make_bucket: lambdas
upload: ../seed/lambdas/pubsub.zip to s3://lambdas/pubsub.zip

An error occurred (InvalidParameterValueException) when calling the CreateFunction operation: Value nodejs10.x at 'runtime' failed to satisfy constraint: Member must satisfy enum value set: [nodejs20.x, provided.al2023, python3.12, python3.13, java17, nodejs16.x, dotnet8, python3.10, java11, python3.11, dotnet6, java21, nodejs18.x, provided.al2, ruby3.3, java8.al2, ruby3.2, python3.8, python3.9] or be a valid ARN
{
    "TopicArn": "arn:aws:sns:us-east-1:000000000000:pubsub"
}
{
    "SubscriptionArn": "arn:aws:sns:us-east-1:000000000000:pubsub:def2382c-8c4728187c40"
}

An error occurred (InvalidParameterValueException) when calling the CreateFunction operation: Value nodejs8.10 at 'runtime' failed to satisfy constraint: Member must satisfy enum value set: [nodejs20.x, provided.al2023, python3.12, python3.13, java17, nodejs16.x, dotnet8, python3.10, java11, python3.11, dotnet6, java21, nodejs18.x, provided.al2, ruby3.3, java8.al2, ruby3.2, python3.8, python3.9] or be a valid ARN
{
    "QueueUrl": "http://sqs.us-east-1.localhost.localstack.cloud:4566/000000000000/user-buffer"
}
{
    "QueueUrl": "http://sqs.us-east-1.localhost.localstack.cloud:4566/000000000000/noti-email"
}
{
    "Location": "/local-media-transcode"
}
{
    "SubscriptionArn": "arn:aws:sns:us-east-1:000000000000:pubsub:6e206a-be10355242e0"
}

An error occurred (InternalError) when calling the CreateTable operation (reached max retries: 2): exception while calling dynamodb.CreateTable: Traceback (most recent call last):
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/rolo/gateway/chain.py", line 166, in handle
    handler(self, self.context, response)
  File "/opt/code/localstack/localstack-core/localstack/aws/handlers/service.py", line 113, in __call__
    handler(chain, context, response)
  File "/opt/code/localstack/localstack-core/localstack/aws/handlers/service.py", line 83, in __call__
    skeleton_response = self.skeleton.invoke(context)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/localstack-core/localstack/aws/skeleton.py", line 154, in invoke
    return self.dispatch_request(serializer, context, instance)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/localstack-core/localstack/aws/skeleton.py", line 168, in dispatch_request
    result = handler(context, instance) or {}
             ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/localstack-core/localstack/aws/forwarder.py", line 133, in _call
    return handler(context, req)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/localstack-core/localstack/aws/skeleton.py", line 118, in __call__
    return self.fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/localstack-core/localstack/aws/api/core.py", line 163, in operation_marker
    return fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/localstack-core/localstack/services/dynamodb/provider.py", line 651, in create_table
    result = self.forward_request(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/localstack-core/localstack/services/dynamodb/provider.py", line 603, in forward_request
    return self.server.proxy(context, service_request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/localstack-core/localstack/aws/forwarder.py", line 59, in __call__
    return self.forward(context, service_request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/localstack-core/localstack/aws/forwarder.py", line 77, in forward
    http_response = self.proxy.forward(context.request, forward_path=context.request.path)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/rolo/proxy.py", line 83, in forward
    proxy_request = _copy_request(request, self.forward_base_url, forward_path, headers)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/rolo/proxy.py", line 149, in _copy_request
    builder = EnvironBuilder.from_environ(request.environ)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/werkzeug/test.py", line 427, in from_environ
    return cls(**out)
           ^^^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/werkzeug/test.py", line 328, in __init__
    base_url = iri_to_uri(base_url)
               ^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/werkzeug/urls.py", line 153, in iri_to_uri
    if parts.port:
       ^^^^^^^^^^
  File "/usr/local/lib/python3.11/urllib/parse.py", line 182, in port
    raise ValueError(f"Port could not be cast to integer value as {port!r}")
ValueError: Port could not be cast to integer value as ' 4566'


An error occurred (InternalError) when calling the CreateTable operation (reached max retries: 2): exception while calling dynamodb.CreateTable: Traceback (most recent call last):
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/rolo/gateway/chain.py", line 166, in handle
    handler(self, self.context, response)
  File "/opt/code/localstack/localstack-core/localstack/aws/handlers/service.py", line 113, in __call__
    handler(chain, context, response)
  File "/opt/code/localstack/localstack-core/localstack/aws/handlers/service.py", line 83, in __call__
    skeleton_response = self.skeleton.invoke(context)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/localstack-core/localstack/aws/skeleton.py", line 154, in invoke
    return self.dispatch_request(serializer, context, instance)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/localstack-core/localstack/aws/skeleton.py", line 168, in dispatch_request
    result = handler(context, instance) or {}
             ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/localstack-core/localstack/aws/forwarder.py", line 133, in _call
    return handler(context, req)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/localstack-core/localstack/aws/skeleton.py", line 118, in __call__
    return self.fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/localstack-core/localstack/aws/api/core.py", line 163, in operation_marker
    return fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/localstack-core/localstack/services/dynamodb/provider.py", line 651, in create_table
    result = self.forward_request(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/localstack-core/localstack/services/dynamodb/provider.py", line 603, in forward_request
    return self.server.proxy(context, service_request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/localstack-core/localstack/aws/forwarder.py", line 59, in __call__
    return self.forward(context, service_request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/localstack-core/localstack/aws/forwarder.py", line 77, in forward
    http_response = self.proxy.forward(context.request, forward_path=context.request.path)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/rolo/proxy.py", line 83, in forward
    proxy_request = _copy_request(request, self.forward_base_url, forward_path, headers)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/rolo/proxy.py", line 149, in _copy_request
    builder = EnvironBuilder.from_environ(request.environ)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/werkzeug/test.py", line 427, in from_environ
    return cls(**out)
           ^^^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/werkzeug/test.py", line 328, in __init__
    base_url = iri_to_uri(base_url)
               ^^^^^^^^^^^^^^^^^^^^
  File "/opt/code/localstack/.venv/lib/python3.11/site-packages/werkzeug/urls.py", line 153, in iri_to_uri
    if parts.port:
       ^^^^^^^^^^
  File "/usr/local/lib/python3.11/urllib/parse.py", line 182, in port
    raise ValueError(f"Port could not be cast to integer value as {port!r}")
ValueError: Port could not be cast to integer value as ' 4566'
```

**Comment by Isanderthul** [comment]
I have no idea if this helps, but this is how I got the same error in a python test.
I had a look and we certainly don't have spaces in the urls of our scripts, so I'm thinking it comes from somewhere in localstack when handing the parts

sampling the `iri_to_uri(iri: str)` from `./.venv/lib/python3.11/site-packages/werkzeug/urls.py`
```test.py
from urllib.parse import urlsplit, urlunsplit, quote

def iri_to_uri(iri: str) -> str:
    """Convert an IRI to a URI. All non-ASCII and unsafe characters are quoted."""
    parts = urlsplit(iri)
    path = quote(parts.path, safe="%!$&'()*+,/:;=@")
    query = quote(parts.query, safe="%!$&'()*+,/:;=?@")
    fragment = quote(parts.fragment, safe="%!#$&'()*+,/:;=?@")

    if parts.hostname:
        netloc = parts.hostname.encode("idna").decode("ascii")
    else:
        netloc = ""

    if ":" in netloc:
        netloc = f"[{netloc}]"

    if parts.port:
        netloc = f"{netloc}:{parts.port}"

    if parts.username:
        auth = quote(parts.username, safe="%!$&'()*+,;=")

        if parts.password:
            password = quote(parts.password, safe="%!$&'()*+,;=")
            auth = f"{auth}:{password}"

        netloc = f"{auth}@{netloc}"

    return urlunsplit((parts.scheme, netloc, path, query, fragment))


# Test the function
if __name__ == "__main__":
    # Example test case
    iri = "http://☃.url.without.space:4556/påth?q=èry%DF"
    uri = iri_to_uri(iri)
    print("Input IRI:", iri)
    print("Output URI:", uri)

    iri = "http://☃.url.with.space: 4556/påth?q=èry%DF"
    uri = iri_to_uri(iri)
    print("Input IRI:", iri)
    print("Output URI:", uri)
```
gives a very similar error
```
Input IRI: http://☃.url.without.space:4556/påth?q=èry%DF
Output URI: http://xn--n3h.url.without.space:4556/p%C3%A5th?q=%C3%A8ry%DF
Traceback (most recent call last):
  File "/Users/dev/Code/localstack/test/test.py", line 42, in <module>
    uri = iri_to_uri(iri)
  File "/Users/dev/Code/localstack/test/test.py", line 18, in iri_to_uri
    if parts.port:
       ^^^^^^^^^^
  File "/Users/dev/.pyenv/versions/3.13.0/lib/python3.13/urllib/parse.py", line 182, in port
    raise ValueError(f"Port could not be cast to integer value as {port!r}")
ValueError: Port could not be cast to integer value as ' 4556'
```

**Comment by Isanderthul** [comment]
OK I have tracked the issue down to the internal company spyware proxy that they run on my work macbook. That spyware proxy is inserting the space when it rewrites urls. 
I addressed this issue in the following way:
Create a local proxy to handle local traffic so that it does not go to the company spyware proxy, and set localstack to use that proxy, using the HTTP_PROXY environment variable.
```
brew install tinyproxy
brew services start tinyproxy
echo 'export HTTP_PROXY=http://localhost:8888' >> ~/.zshrc
```
